server/src/main/java/org/elasticsearch/index/mapper/FieldTypeLookup.java

```diff
@@ -147,6 +150,19 @@ private MappedFieldType getDynamicField(String field) {
         }
     }
 
+    Collection<MappedFieldType> getMatchingFieldTypes(Predicate<MappedFieldType> predicate) {
+        if (dynamicFieldTypes.isEmpty()) {
+            return fullNameToFieldType.values().stream().filter(predicate).collect(Collectors.toList());
+        }
+        return Stream.concat(dynamicFieldTypes.values().stream().flatMap(dft -> {
+            List<MappedFieldType> childFieldTypes = new ArrayList<>();
+            for (String knownSubfield : dft.getKnownSubfields()) {
+                childFieldTypes.add(dft.getChildFieldType(knownSubfield));
+            }
+            return childFieldTypes.stream();
```

Comment:
I am thinking that probably, given the current usages of this method, the dynamic field type step can be skipped. `getKnownSubfields` was just introduced for #73252, and eager global ordinals and join field type (where this method is called) have nothing to do with runtime fields.